File: Advent-code2022/day1.py
import sys
total = 0
amounts = []
for a in sys.stdin:
    print()
    if a == "\n":
        print("newline")
        amounts.append(total)
        total = 0
    else:
        total = total + int(a)

amounts.sort(reverse=True)
print(amounts)
print("end")
# Day 1
print(max(amounts))
print(amounts[0], amounts[1], amounts[2])
# Day 2
print(amounts[0] + amounts[1] + amounts[2])

File: Advent-code2022/day1.cc
#include <bits/stdc++.h>
using namespace std;

typedef long long ll;
const double PI = acos(-1.0);
#define INF 1000000000


int main() {
    string line;
    vector<int> amounts;
    int current = 0;
    while(getline(cin, line)) {
        if(line == "") {
            cout << "newline" << endl;
            amounts.push_back(current);
            current = 0;
        }
        else {
            int temp = stoi(line);
            //cout << typeof(temp) << endl;
            //cout << temp << endl;
            current += temp;
            cout << current << endl;
        }
    }
    for(int i=0; i<amounts.size(); i++) {
        cout << amounts[i] << endl;
    }

    cout << "end" << endl;
    sort(rbegin(amounts), rend(amounts));
    // int max = 0;
    // for(auto i : amounts) {
    //     if(i >= max) {
    //         max = i;
    //     }
    // }
    // auto it = max_element(begin(amounts), end(amounts));
    // cout << amounts[it]
    //std::cout << *std::max_element(begin(amounts), end(amounts))  << endl;
    cout << amounts[0] << " " << amounts[1] << " " << amounts[2] << endl;
    cout << amounts[0] + amounts[1] + amounts[2] << endl;
  return 0;
}